soc/nordic/common/uicr/gen_uicr.py

```diff
@@ -197,6 +204,32 @@ def main() -> None:
             uicr.PERIPHCONF.ADDRESS = periphconf_address
             uicr.PERIPHCONF.MAXCOUNT = math.floor(periphconf_size / 8)
 
+        # Configure RECOVERY field if enabled
+        if kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY") == "y":
+            uicr.RECOVERY.ENABLE = ENABLED_VALUE
+
+            # Determine processor core from choice configuration
+            if kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_PROCESSOR_APPLICATION") == "y":
+                processor = UICRV2_RECOVERY_PROCESSOR_PROCESSOR_APPLICATION
+            elif kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_PROCESSOR_RADIOCORE") == "y":
+                processor = UICRV2_RECOVERY_PROCESSOR_PROCESSOR_RADIOCORE
+            else:
+                raise ScriptError("Unreachable code")
+            uicr.RECOVERY.PROCESSOR = processor
+
+            # Get and parse INITSVTOR address
+            initsvtor = int(kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_INITSVTOR"), 0)
+            uicr.RECOVERY.INITSVTOR = initsvtor
+
+            # Get and parse size in 4KB blocks
+            size4kb = int(kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_SIZE4KB"), 0)
+            uicr.RECOVERY.SIZE4KB = size4kb
+
+            if kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_PERIPHCONF") == "y":
+                uicr.RECOVERY.PERIPHCONF.ENABLE = ENABLED_VALUE
+                uicr.RECOVERY.PERIPHCONF.ADDRESS = periphconf_address
+                uicr.RECOVERY.PERIPHCONF.MAXCOUNT = math.floor(periphconf_size / 8)
```

Comment:
I think we should at least make this a copy of the periphconf data at a different location instead of pointing to the same location - also defining a separate partition like periphconf_recovery_partition in the board dts memory map for it. Then it is simpler to improve it to generate data for these two partitions from separate sources later

Reply:
fixed